sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -834,4 +834,21 @@ object IntervalUtils {
    * @return The period of months, not null
    */
   def monthsToPeriod(months: Int): Period = Period.ofMonths(months).normalized()
+
+  /**
+   * Converts an year-month interval as a number of months to its textual representation
+   * which conforms to the ANSI SQL standard.
+   *
+   * @param months The number of months, positive or negative
+   * @return Year-month interval string
+   */
+  def toYMIntervalString(months: Int): String = {
+    var sign = ""
+    var absMonths: Long = months
+    if (months < 0) {
+      sign = "-"
+      absMonths = -absMonths
+    }
+    s"interval '$sign${absMonths / MONTHS_PER_YEAR}-${absMonths % MONTHS_PER_YEAR}' year to month"
```

Comment:
The last part "year to month" looks a bit confusing to me.  What does other DBMS systems do?

Reply:
> The last part "year to month" looks a bit confusing to me

This part conforms to the SQL standard, see:
```
<interval literal> ::=
  INTERVAL [ <sign> ] <interval string> <interval qualifier>

<interval qualifier> ::=
    <start field> TO <end field>
  | <single datetime field>
<start field> ::=
<non-second primary datetime field>
      [ <left paren> <interval leading field precision> <right paren> ]
<end field> ::=
<non-second primary datetime field>
  | SECOND [ <left paren> <interval fractional seconds precision> <right paren> ]
<single datetime field> ::= <non-second primary datetime field>
        [ <left paren> <interval leading field precision> <right paren> ]
  | SECOND [ <left paren> <interval leading field precision>
      [ <comma> <interval fractional seconds precision> ] <right paren> ]
<primary datetime field> ::= <non-second primary datetime field>
| SECOND
<non-second primary datetime field> ::= YEAR
  | MONTH
  | DAY
  | HOUR
  | MINUTE
```

> What does other DBMS systems do?

For instance, see Oracle https://docs.oracle.com/en/database/oracle/oracle-database/12.2/sqlrf/Literals.html#GUID-4C258D8F-3DF2-4D45-BE3E-14864DD77100

Reply:
I see, thanks for the explanation!

Reply:
As per the standard, it seems the output of "-13 months" should be 
```interval - '1-1' year to month```
instead of
```interval '-1-1' year to month```

Reply:
Both are allowed by the standard. See the grammar rules:
```
<unquoted interval string> ::=
[ <sign> ] { <year-month literal> | <day-time literal> }
```
```
<interval literal> ::=
  INTERVAL [ <sign> ] <interval string> <interval qualifier>
```

Reply:
I see. Shall we add one space in between the sign and interval string as per the standard?

Reply:
I think the space doesn't matter. See for example:
<img width="769" alt="Screenshot 2021-04-06 at 12 00 12" src="https://user-images.githubusercontent.com/1580697/113686315-cd32fe80-96cf-11eb-83e5-12b42e78e9c0.png">